### PR TITLE
chore: Allow to configure resources for the post-install-hook pod

### PR DESCRIPTION
**Pull request by marcofranssen**
**Description**

This allows to define the postInstallHook resources so policies on our cluster can be met that require us to define resources for any container.

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: charts/karpenter/values.yaml [hunk 1]
```diff
@@ -145,6 +145,17 @@ postInstallHook:
     tag: "1.30"
     # -- SHA256 digest of the post-install hook image.
     digest: sha256:13a2ad1bd37ce42ee2a6f1ab0d30595f42eb7fe4a90d6ec848550524104a1ed6
+  # -- Resources for the post-install-hook pod.
+  resources: {}
```
Comment (jonathan-innis) [inline charts/karpenter/values.yaml:149]: I can't imagine that this is going to be variable. Can we just set this to static resource values and not need configurable values?

Comment (marcofranssen) [reply charts/karpenter/values.yaml:149]: This is a best practice. Resources are never hardcoded.

E.g. Depending on a specific cluster the requirements for resources might differ. One cluster might just have 1 nodepool where another one might have many more where more resources might be allocated to perform the task.

File: charts/karpenter/templates/post-install-hook.yaml [hunk 2]
```diff
@@ -26,20 +26,23 @@ spec:
...
+                kubectl patch customresourcedefinitions ec2nodeclasses.karpenter.k8s.aws  --type='merge' -p '{"spec":{"conversion":{"strategy": "Webhook", "webhook":{"conversionReviewVersions": ["v1beta1", "v1"], "clientConfig":{"service":{"name":"{{ include "karpenter.fullname" . }}", "port": {{ .Values.webhook.port }} ,"namespace": "{{ .Release.Namespace }}"}}}}}}'
+              {{- else }}
+                echo "disabled webhooks"
+                kubectl patch customresourcedefinitions nodepools.karpenter.sh  --type='json' -p '[{'op': 'remove', 'path': '/spec/conversion'}]'
+                kubectl patch customresourcedefinitions nodeclaims.karpenter.sh  --type='json' -p '[{'op': 'remove', 'path': '/spec/conversion'}]'
+                kubectl patch customresourcedefinitions ec2nodeclasses.karpenter.k8s.aws  --type='json' -p '[{'op': 'remove', 'path': '/spec/conversion'}]'
+              {{- end }}
+          {{- with .Values.controller.resources }}
```
Comment (marcofranssen) [inline charts/karpenter/templates/post-install-hook.yaml:45]:
```suggestion
          {{- with .Values.postInstallHook.resources }}
```

**Comment by engedaam** [comment]
Closing as the post install hooks have been removed

**Pull request closed without merging**